src/pull.rs

```diff
@@ -1,14 +1,32 @@
 use anyhow::Result;
+use indicatif::{ProgressBar, ProgressStyle};
 use policy_evaluator::policy_fetcher::{
     fetch_policy, policy::Policy, sources::Sources, PullDestination,
 };
+use std::time::Duration;
 
 pub(crate) async fn pull(
     uri: &str,
     sources: Option<&Sources>,
     destination: PullDestination,
 ) -> Result<Policy> {
-    fetch_policy(uri, destination, sources)
+    let pb = ProgressBar::new_spinner();
+    pb.set_style(
+        ProgressStyle::default_spinner()
+            .template("{spinner:.green} {msg}")
+            .unwrap(),
```

Comment:
nitpick: I prefer to not have `unwrap` inside of prod code. In this case the code should never fail, hence it's fine to use an `expect` instead of raising the error

```suggestion
            .expect("cannot set spinner template"),
```